### DISCUSSION TITLE
What would be the best workflow for creating a stock forecasting model?

### DISCUSSION THREAD

**Discussion by whoopydev** [autogluon/autogluon#2609] [Q&A]
Hey guys, 
For a university project I have decided to create a stock forecasting model which can perform real-time inference, and I'm wondering what would be the best way to go about this. I have some questions and I'd be extremely grateful if I could get any guidance as to how to go about them. 

1)
Would you guys recommend training one model with all kinds of financial data (I'm talking data for thousands of stocks and market indices) or would you recommend training one model for each stock? 

The obvious answer obviously is training one model with all different kinds of data, however I'm wondering what exactly would the workflow look like (I'm relatively new to machine learning and time series forecasting). 

2) 
The timeforecasting documentations says the following : 
"_For time-series data containing multiple individual series, AutoGluon can produce forecasting models to predict future values of each series based on historical observations of both this series and all of the other series in the dataset. A single call to AutoGluon TimeSeriesPredictor’s fit() automatically trains multiple models on a dataset containing multiple time-series measured over the same time period, and does not require you to manually deal with cumbersome issues like data cleaning, hyperparameter optimization, model selection, etc."_

Here is says it can look at data from many different timeseries and use that to predict future values of any of the series. How would I go about doing this if I had many hundreds of different series in individual CSV files? In certain ML libraries such as Darts, stacking timeseries is an option, however stacking so many series doesn't seem like an efficient way to about things. 

3) 
How can I perform real-time inference to get predictions for dates of the future which aren't in my dataset? 

4) 
Assuming I manage to train a model with hundreds of different time series, I assume I will continuously need to update the data of each series every day with the latest stock prices if I want to get accurate next-day predictions. Assuming I have to do this, will I need to retrain the entire model on a daily basis or is there a way for me to simply perform inference with the latest data without having to train the model again? 

Once again, any help would be greatly appreciated! Thanks!

**Comment by alfa14290** [comment]
hey did you find the answers for this question because I am struggling with the same questions. Would be kind if yu can let me know

**Comment by dshetyo** [comment]
Following
